api/controllers/kosh.js

```diff
@@ -45,9 +45,59 @@ exports.word = async (req, res) => {
   }
 };
 
+exports.wordSearch = async (req, res) => {
+  let conn;
+  const query = `${req.params.query}`;
+  const match = `%${query}%`;
+  const fullMatch = `${query}`;
+  const startMatch = `${query}%`;
+  const endMatch = `%${query}`;
+  try {
+    conn = await req.app.locals.pool.getConnection();
+    const q = `SELECT w.ID AS id, w.Word AS word, w.WordUni AS wordUni,
+                  d.DefGurmukhi AS definition, d.DefGurmukhiUni AS definitionUni
+                FROM MahanKoshWords w
+                LEFT JOIN MahanKoshDefinitions d ON w.Definition = d.ID
+                WHERE
+                  w.Word LIKE ? OR
+                  w.WordUni LIKE BINARY ?
+                ORDER BY 
+                  CASE
+                    WHEN word LIKE ? THEN 1
+                    WHEN wordUni LIKE ? THEN 1
+                    WHEN word LIKE ? THEN 2
+                    WHEN wordUni LIKE ? THEN 2
+                    WHEN word LIKE ? THEN 3
+                    WHEN wordUni LIKE ? THEN 3
+                    ELSE 4
+                  END`;
+    const rows = await conn.query(q, [
+      match,
+      match,
+      match,
+      match,
+      fullMatch,
+      fullMatch,
+      startMatch,
+      startMatch,
+      endMatch,
+      endMatch,
```

Comment:
Am I reading this correctly? There are 10 arguments being passed but only 8 `?`s in the query

Reply:
I think maybe there should only be two `match` values included in the arguments array, yes?

Reply:
good catch. copy/paste error slipped through.

Reply:
Will prioritize #113 next so these things can be caught earlier.